docs/topics/unused-return-value-checker.md

```diff
@@ -0,0 +1,173 @@
+This helps prevent unexpected behavior and makes such issues easier to track down.
+
+Here's an example:
+
+```kotlin
+fun formatGreeting(name: String): String {
+    if (name.isBlank()) return "Hello, anonymous user!"
+    if (!name.contains(' ')) {
+        // The checker reports a warning that this result is ignored
+        "Hello, " + name.replaceFirstChar(Char::titlecase) + "!"
+        // Unused return value of 'plus'.
+    }
+    val (first, last) = name.split(' ')
+    return "Hello, $first! Or should I call you Dr. $last?"
+}
+```
+
+In this example, a string is created but never used, so the checker reports it as an ignored result.
+
+## Configure the unused return value checker
+
+You can control how the compiler reports ignored results with the `-Xreturn-value-checker` compiler option.
+
+It has the following modes:
+
+* `disable` disables the unused return value checker. (Default)
+* `check` reports warnings for ignored results from [marked functions](#mark-functions-to-check-ignored-results).
+* `full` reports warnings for ignored results from all functions in your project.
+
+To use the unused return value checker in your project, add the compiler option to your build configuration file:
+
+<tabs>
+<tab id="kotlin" title="Gradle">
+
+```kotlin
+// build.gradle(.kts)
+kotlin {
+    compilerOptions {
+        freeCompilerArgs.add("-Xreturn-value-checker=check")
+    }
+}
+```
+</tab>
+
+<tab id="maven" title="Maven">
+
+```xml
+<!-- pom.xml -->
+<plugin>
+    <groupId>org.jetbrains.kotlin</groupId>
+    ..
+    <configuration>
+        <args>
+            <arg>-Xreturn-value-checker=check</arg>
+        </args>
+    </configuration>
+</plugin>
+```
+
+</tab>
+</tabs>
+
+## Mark functions to check ignored results
+
+When you set [the `-Xreturn-value-checker` compiler option](#configure-the-unused-return-value-checker) to `check`,
+the checker reports ignored results only from expressions that are marked, like most functions in the Kotlin standard library.
+
+To mark your own code, 
+use the [`@MustUseReturnValues`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin/-must-use-return-value/) annotation.
+You can apply it to a file, class, or function depending on the scope you want the checker to cover.
+
+For example, you can mark an entire file:
+
+```kotlin
+// Marks all functions and classes in this file so the checker reports unused return values
+@file:MustUseReturnValues
+
+package my.project
+
+fun someFunction(): String
+```
+
+Or a specific class:
+
+```kotlin
+// Marks all functions in this class so the checker reports unused return values
+@MustUseReturnValues
+class Greeter {
+    fun greet(name: String): String = "Hello, $name"
+}
+
+fun someFunction(): Int = ...
+```
+
+To apply the checker to your entire project, set the `-Xreturn-value-checker=check` compiler option to `full`.
```

Comment:
```suggestion
To apply the checker to your entire project, set the `-Xreturn-value-checker` compiler option to `full`.
```

Can we highlight this paragraph somehow? Using `full` mode should be the preferrable choice for users instead of `@MustUseReturnValues`